importer/balanced/builder.go

```diff
@@ -130,6 +130,49 @@ import (
 //        +=========+   +=========+   + - - - - +
 //
 func Layout(db *h.DagBuilderHelper) (ipld.Node, error) {
+	if !db.IsMultiDagBuilder() {
+		return layout(db)
+	}
+
+	dbs := db.MultiHelpers()
+	type ne struct {
+		n   ipld.Node
+		err error
+	}
+	dbc := make(chan ne)
+	// Build sub trees concurrently
+	for _, dbh := range dbs {
+		go func(db *h.DagBuilderHelper) {
```

Comment:
The current PR lgtm 
except this one. 
I think we can check the followings and have a sort of counting semaphore to limit # of concurrent go routines: 
GOMAXPROCS (if still a hard limit in golang 1.12 or later?), heap memory size limit and FD (file descriptor) limit 
of the current process (if we can not change in runtime), and physical memory of the current system. 
But I am OK to merge this PR without addressing this, but I think we need to address before beta.

Reply:
The number of `len(dbs)`, or the number of goroutines here is limited, see https://github.com/TRON-US/go-btfs-chunker/blob/master/parse.go#L117

I think the real issue you mentioned is the FD limit... which we will need to address on large files, but I think the fix will be on a top level at coreunix/coreapi. I will have a ticket to track this. Thanks.

Reply:
I think we can address with another ticket regarding other system limits that I mentioned. 
Each node/machine may have a different set of such limit values depending on its capacity. 

What I am suggesting is to adjust the controllable numbers (like # of shards, in the current code, the threshold is hard-coded as 256, also # of concurrent go routines which can be controlled via a simple bounded task queue or a counting semaphone) according to the variable size of the node/machine. 

But again, I am OK if you have a ticket.

Reply:
I am not comfortable with limiting # of goroutines. One of Golang's major advantages is the scheduling of goroutines based on compiled code and resource, limiting it would chip away that advantage.

Reply:
OK.. we can discuss later. But controlling Degree of Parallelism via queue or semaphore is common. For example, when 5 go routines requir 5 * number-of-file-descripters-used-by-one-go-routine and, if this is beyond the process FD limit, we can limit the # of concurrent go routines. This can be applied other resources like memory, # of CPUs, etc.